task_processing/plugins/mesos/execution_framework.py

```diff
@@ -106,28 +112,45 @@ def __init__(
 
         self.stopping = False
         task_kill_thread = threading.Thread(
-            target=self.kill_tasks_stuck_in_staging, args=())
+            target=self._background_check, args=())
         task_kill_thread.daemon = True
         task_kill_thread.start()
 
-    def kill_tasks_stuck_in_staging(self):
+    def _background_check(self):
         while True:
             if self.stopping:
                 return
 
+            time_now = time.time()
             with self._lock:
-                time_now = time.time()
                 for task_id in self.task_metadata.keys():
-                    if self.task_metadata[task_id].task_state not in (
+                    md = self.task_metadata[task_id]
+
+                    if md.task_state not in (
                         'TASK_STAGING',
```

Comment:
this isn't a pattern introduced by you, but these magic strings kind of bug me. We should define constants for these things.

Reply:
Yeah, I agree. Created a separate ticket: TASKPROC-122